index.js

```diff
@@ -111,6 +113,111 @@ async function newAllReconciliations(firmId) {
+  const template = await SF.readExportFileById(firmId, id);
+  if (!template) {
+    throw `Export file with id ${id} wasn't found`;
+  }
+  ExportFile.save(firmId, template);
+}
+
+async function fetchAllExportFiles(firmId, page = 1) {
+  const templates = await SF.readExportFiles(firmId, page);
+  if (templates.length == 0) {
+    if (page == 1) {
+      console.log("No export files found");
+    }
+    return;
+  }
+  templates.forEach(async (template) => {
+    fetchExportFileById(firmId, template.id);
+  });
+  fetchAllExportFiles(firmId, page + 1);
+}
+
+async function publishExportFileByName(
+  firmId,
+  name,
+  message = "Updated through the API"
+) {
+  try {
+    const templateConfig = fsUtils.readConfig("exportFile", name);
+    if (!templateConfig || !templateConfig.id[firmId]) {
+      errorUtils.missingExportFileId(name);
+    }
+    let templateId = templateConfig.id[firmId];
+    console.log(`Updating export file ${name}...`);
+    const template = await ExportFile.read(name);
+    template.version_comment = message;
+    const response = await SF.updateExportFile(firmId, templateId, template);
+    if (response && response.data && response.data.name) {
+      console.log(`Export file updated: ${response.data.name}`);
+      return true;
+    } else {
+      console.log(`Export file update failed: ${name}`);
+      return false;
+    }
+  } catch (error) {
+    errorUtils.errorHandler(error);
+  }
+}
+
+async function publishAllExportFiles(
+  firmId,
+  message = "Updated through the API"
+) {
+  let templates = fsUtils.getAllTemplatesOfAType("exportFile");
+  for (let name of templates) {
+    if (!name) continue;
+    await publishExportFileByName(firmId, name, message);
+  }
+}
+
+async function newExportFile(firmId, name) {
+  try {
+    const existingTemplate = await SF.findExportFileByName(firmId, name);
+    if (existingTemplate) {
+      console.log(
+        `Reconciliation ${name} already exists. Skipping its creation`
+      );
+      return;
+    }
+    const template = await ExportFile.read(name);
+    template.version_comment = "Created through the API";
+    const response = await SF.createExportFile(firmId, template);
+
+    // Store new id
+    if (response && response.status == 201) {
+      ExportFile.updateTemplateId(firmId, name, response.data.id);
+    }
+  } catch (error) {
+    errorUtils.errorHandler(error);
+  }
+}
+
+async function newAllExportFiles(firmId) {
+  const templates = fsUtils.getAllTemplatesOfAType("exportFile");
+  for (let name of templates) {
+    await newExportFile(firmId, name);
+  }
+}
+
+async function fetchAccountDetailByHandle(firmId, name) {}
+async function fetchAccountDetailById(firmId, id) {}
+async function fetchAllAccountDetails(firmId) {}
+async function publishAccountDetailByName(firmId, name) {}
+async function publishAllAccountDetails(firmId) {}
+async function newAccountDetail(firmId, name) {}
+async function newAllAccountDetails(firmId) {}
```

Comment:
Should we leave these as part of this PR?

Reply:
Removed